apps/server/lib/lexical/server/code_intelligence/completion/translations/macro.ex

```diff
@@ -6,13 +6,62 @@ defmodule Lexical.Server.CodeIntelligence.Completion.Translations.Macro do
 
   use Translatable.Impl, for: Result.Macro
 
-  @snippet_macros ~w(def defp defmacro defmacrop defimpl defmodule defprotocol defguard defguardp defexception)
+  @snippet_macros ~w(def defp defmacro defmacrop defimpl defmodule defprotocol defguard defguardp defexception test)
 
   def translate(%Result.Macro{name: name}, _builder, _env)
       when name in ["__before_compile__", "__using__", "__after_compile__"] do
     :skip
   end
 
+  def translate(%Result.Macro{name: "test", arity: 1} = macro, builder, env) do
+    label = "#{macro.name} (Define an empty test/1)"
+
+    snippet = """
+    test "${0:name}"
+    """
+
+    builder.snippet(env, snippet,
+      detail: macro.spec,
+      kind: :class,
+      label: label,
+      sort_text: builder.boost(label, 10)
+    )
+  end
+
+  def translate(%Result.Macro{name: "test", arity: 2} = macro, builder, env) do
+    label = "#{macro.name} (Define test/2)"
+
+    snippet = """
+    test "${1:name}" do
+      $0
+    end
+    """
+
+    builder.snippet(env, snippet,
+      detail: macro.spec,
```

Comment:
I do not know. I found that macro struct is defined here https://github.com/lexical-lsp/lexical/blob/f45d0ccb0d72a87c532158525af4959a5b2ae3a6/apps/remote_control/lib/lexical/remote_control/completion/candidate.ex#L40 

I also found that the `macro` var in this function seems to come from this `local_completions` here https://github.com/lexical-lsp/lexical/blob/f45d0ccb0d72a87c532158525af4959a5b2ae3a6/apps/server/lib/lexical/server/code_intelligence/completion.ex#L73

But... I still do not know if `macro` has a `spec` in this scope, how can I find out?

Reply:
does the spec come from elixir_sense? 
I chased `local_completions` and landed here:
https://github.com/lexical-lsp/lexical/blob/f45d0ccb0d72a87c532158525af4959a5b2ae3a6/apps/remote_control/lib/lexical/remote_control/api.ex#L44

Reply:
So I ran the code and I see this:

```
02:21:52.675 [info] Local completions are [

%LXRelease.RemoteControl.Completion.Candidate.Function{argument_names: ["tuple"], arity: 1, name: "tuple_size", origin: "Kernel", type: :function, visibility: :public, spec: "@spec tuple_size(tuple) :: non_neg_integer", metadata: %{guard: true}}, 

%LXRelease.RemoteControl.Completion.Candidate.Macro{argument_names: ["value", "fun"], arity: 2, name: "then", origin: "Kernel", type: :macro, visibility: :public, spec: "", metadata: %{since: "1.12.0"}}, 

%LXRelease.RemoteControl.Completion.Candidate.Macro{argument_names: ["message"], arity: 1, name: "test", origin: "ExUnit.Case", type: :macro, visibility: :public, spec: "", metadata: %{}}, 

%LXRelease.RemoteControl.Completion.Candidate.Macro{argument_names: ["message", "contents"], arity: 2, name: "test", origin: "ExUnit.Case", type: :macro, visibility: :public, spec: "", metadata: %{defaults: 1}}, 

%LXRelease.RemoteControl.Completion.Candidate.Macro{argument_names: ["message", "var", "contents"], arity: 3, name: "test", origin: "ExUnit.Case", type: :macro, visibility: :public, spec: "", metadata: %{defaults: 1}}]
```

so you can see some structs in there that have `name: "test"` and `spec: ""`.

So it is indeed empty!

So I guess I just delete the `detail: macro.spec` field, correct?